iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp

```diff
@@ -215,18 +222,56 @@ static bool haveEqualShapeDim(Value x, Value y, int i) {
          y.getType().cast<ShapedType>().getDimSize(i);
 }
 
+// Current instances of integer matmuls in linalg tend to use signless integers
+// to mean signed. Accept both, while this is sorted out.
+static bool isSignedOrSignlessInteger(Type type, int width) {
+  return type.isSignedInteger(width) || type.isSignlessInteger(width);
+}
+
+class Mmt4DTileParams {
+public:
+  Mmt4DTileParams(int64_t M0, int64_t K0, int64_t N0, std::string comment) : M0(M0), K0(K0), N0(N0), comment(comment) {}
+  std::array<int64_t, 2> lhs() const { return {M0, K0}; }
+  std::array<int64_t, 2> rhs() const { return {K0, N0}; }
+  std::array<int64_t, 2> acc() const { return {M0, N0}; }
+  const std::string& getComment() const { return comment; }
+private:
+  const int64_t M0;
+  const int64_t K0;
+  const int64_t N0;
+  const std::string comment;
+};
+
+static Mmt4DTileParams chooseTileParamsForElemTypesAndTargetInfo(Type lhsElemType, Type rhsElemType, Type accElemType, const ConvertMatmulToMmt4DTargetInfo& target_info)
```

Comment:
Seems like this could just be "chooseTileParams" the parameters to the function tell us what it is choosing them based on

Reply:
this is just a local helper for the actual `chooseTileParams` function just below. i thought that being a helper made it a non-issue that the name was verbose, and that the verbosity made the call site logic easier to parse. At the momet it's probably confusing because the call site doesn't do anything else but the TODO there explains how this is expected to diverge, and i thought there was value in documenting the plan in the code here, LMK.

Reply:
Overhauled.